controllers/ovndbcluster_controller.go

```diff
@@ -749,6 +751,12 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 		if err != nil {
 			return ctrl.Result{}, err
 		}
+		// It can be possible that not all pods are ready, so DNSData won't
```

Comment:
Trying to understand why can't we just move this length check below at L723 and avoid/remove the minLenth calculator?

Reply:
Yes sorry, i meant "above". Not sure how much that helps to create DNSData for partial ip list. But just wanted to understand the reasoning for that, as this would just lead to unnecessary respawns for DNSMasq pod and if could avoid the hacks around Min checks. But if we still want to go with this i am ok

Reply:
It will respawn once new pods show up, yes. That's sadly how dnsmasq is implemented right now.

Reply:
Yeah, I was not sure about the behavior regarding the reconcile when the DNSData doesn't change.
My concern was in the scenario where replicas > len(podList), that the reconcile will be triggered until condition is fulfilled, and in every reconcile loop (with the current code) the [DNSData creation will be called](https://github.com/openstack-k8s-operators/ovn-operator/blob/main/controllers/ovndbcluster_controller.go#L741), and I was afraid that it would trigger the recreation of the dnsmasq pod everytime. But that only will happen if the DNSData has different information.

I don't think there's much of an overhead regarding not waiting until all pods are ready. Having said that, I'm okay with any of the two solutions. 
WDYT?

Reply:
Comments added